### PR TITLE
Skip unnecessary share retrieval

### PR DESCRIPTION
**Pull request by butonic**
AFAICT we already got a list of all reveiced shares, including the Hidden flag. No need to make another GetReceivedShare request for every accepted share.

### CODE REVIEW

File: internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go [hunk 1]
```diff
@@ -95,7 +95,6 @@ func (h *Handler) AcceptReceivedShare(w http.ResponseWriter, r *http.Request) {
 			}
 		} else {
 			if s.State == collaboration.ShareState_SHARE_STATE_ACCEPTED {
-				s.Hidden = h.getReceivedShareHideFlagFromShareID(r.Context(), shareID)
```
Comment (kobergj) [inline internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go:98]: We do not need the hidden flag any more?

Do we still need `getReceivedShareHideFlagFromShareID`?

**Review by kobergj — CHANGES_REQUESTED** [review]

File: internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go (hunk 1 above)
Comment (kobergj) [reply internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go:98]: I mean: Don't we need to set the flag somewhere else?

Comment (butonic) [reply internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go:98]: there were two other places ... but both were actually unnecessary as well. we should have done a better review of the PRs ... 😞  https://github.com/cs3org/reva/pull/4454/commits/19b6b46043232bc0f0dd7b72c677fd1a45d35cfb

File: internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go [hunk 2]
```diff
@@ -166,15 +173,15 @@ func (h *Handler) RejectReceivedShare(w http.ResponseWriter, r *http.Request) {
 		Share: &collaboration.Share{
 			Id: &collaboration.ShareId{OpaqueId: shareID},
 		},
-		State:  collaboration.ShareState_SHARE_STATE_REJECTED,
-		Hidden: h.getReceivedShareHideFlagFromShareID(r.Context(), shareID),
+		State: collaboration.ShareState_SHARE_STATE_REJECTED,
 	}
-	updateMask := &fieldmaskpb.FieldMask{Paths: []string{"state", "hidden"}}
+	updateMask := &fieldmaskpb.FieldMask{Paths: []string{"state"}}
```
Comment (butonic) [inline internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go:178]: we need to make sure the share managers respect the field mask!

Comment (butonic) [reply internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go:178]: jsonsc3 handles it, memory handles it, owncloudsql handles it (but does not know anything about the hidden flag), json handles it (but does not know anything about the hidden flag), cs3 also has it. good to merge

**Review by kobergj — APPROVED** [review]

**Review by kobergj — APPROVED** [review]
Looks good to me, but please run a full-ci on ocis before merging. These sharing changes can have unforseen consequences.

**Review by dragonchaser — APPROVED** [review]

**Pull request merged**